playground/src/main.tsx

```diff
@@ -4,11 +4,43 @@ import Editor from "./Editor";
 import "./index.css";
 import { loader } from "@monaco-editor/react";
 import { setupMonaco } from "./Editor/setupMonaco";
+import { restore, stringify } from "./Editor/settings";
+import { DEFAULT_PYTHON_SOURCE } from "./constants";
+import init, { Workspace } from "./ruff";
 
-loader.init().then(setupMonaco);
+startPlayground()
+  .then(({ sourceCode, settings, ruffVersion }) => {
+    ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
+      <React.StrictMode>
+        <Editor
+          initialSettings={settings}
+          initialSource={sourceCode}
+          ruffVersion={ruffVersion}
+        />
+      </React.StrictMode>,
+    );
+  })
+  .catch((error) => console.error("Failed to start playground", error));
```

Comment:
Doesn't this mean that nothing on the page will render until after the data fetch?

Reply:
But, we don't even show the editor until the source is loaded, right?

Reply:
We did show the editor. See https://github.com/astral-sh/ruff/pull/12659/files#r1704097956

Isn't that changing this is your main concern?

Reply:
I thought we showed the chrome (header) but not the editable panes in the editor, just based on reading the code where we have that `source ?` ternary in the editor body.

Reply:
ah yeah, that's correct.

Reply:
I defer to you, I don't want to get in your way.

Reply:
I extracted a `Chrome` component. It would have been nice if [`React.use` ](https://react.dev/reference/react/use) weren't experimental :(